ui/src/utils/QualityControl.js

```diff
@@ -40,11 +40,45 @@ export class QualityControl {
             errorMessage: '',
         };
 
+        const audioResult = this.soundCheck();
+
         if (this.audioBuffer.duration < 2.0) {
             qualityResult.success = false;
             qualityResult.errorMessage += 'Audio recording failed: recording was too short. Try again';
```

Comment:
might want to just have "failure cases" instead so the components using this util can handle how it looks to the user
like, instead of `errorMessage`s do something like: 
```
qualityResult =  {
    success: Boolean
    failureCase: [ tooShort || tooQuiet ]
}```

so it can be easily parsed by the component

Reply:
I'm not sure you need the success boolean since undefined or null could signify success, but it's up to you. It could be helpful to have an error list.
const errors = { TOO_SHORT: 'too short message', TOO_LONG: 'too long message' }